File: closures.php
#!/usr/bin/env php
<?php

function getDecision($name, $game, $answer) {
  echo "$name : $game : $answer \n";
}

function main() {

  $my_game = "my game";
  $my_answer = "my answer";


  $getDecisionCaller = function($name) use ($my_game, $my_answer) {
    getDecision($name, $my_game, $my_answer);
  };

  // Dont need this line, can just use: $getDecisionCaller('company 0');
  getDecision('company 0', $my_game, $my_answer);

  $getDecisionCaller('company 1');
  $getDecisionCaller('company 2');

}

main();

?>

File: data/snippets/github.com/khilnani/snippets/closures.py
#!/usr/bin/env python

def getDecision(name, game, answer):
  print name, game, answer


my_game = 'my game'
my_answer = 'my answer'

getDecision('company 0', my_game, my_answer)


def make_decision(game, answer):
  def callDecision (name):
    getDecision(name, game, answer)
  return callDecision

caller = make_decision(my_game, my_answer)
caller('company 1')
caller('company 2')
